zilliqa/src/contracts/deposit.sol

```diff
@@ -311,6 +324,30 @@ contract Deposit {
         return _stakersMap[blsPubKey].rewardAddress;
     }
 
+    function getControlAddress(
+        bytes calldata blsPubKey
+    ) public view returns (address) {
+        require(blsPubKey.length == 48);
+        if (_stakersMap[blsPubKey].controlAddress == address(0)) {
+            revert("not staked");
+        }
+        return _stakersMap[blsPubKey].controlAddress;
+    }
+
+    function setRewardAddress(
+        bytes calldata blsPubKey,
+        address rewardAddress
+    ) public onlyControlAddress(blsPubKey) {
+        _stakersMap[blsPubKey].rewardAddress = rewardAddress;
+    }
+
+    function setControlAddress(
+        bytes calldata blsPubKey,
+        address controlAddress
+    ) public onlyControlAddress(blsPubKey) {
+        _stakersMap[blsPubKey].controlAddress = controlAddress;
```

Comment:
In future we could consider 2-step control transfer to avoid transferring control to an address that isn't callable (e.g. because no one has the secret key).